packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,345 @@
+
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import { Construct, Duration, IResource, Resource, Stack } from '@aws-cdk/core';
+import { CfnJobDefinition } from './batch.generated';
+
+/**
+ * Properties of a job definition container
+ */
+export interface IJobDefinitionContainer {
+    /**
+     * The command that is passed to the container.
+     *
+     * If you provide a shell command as a single string, you have to quote command-line arguments.
+     *
+     * @default - CMD value built into container image.
+     */
+    readonly command?: string[];
+
+    /**
+     * The environment variables to pass to the container.
+     *
+     * @default none
+     */
+    readonly environment?: { [key: string]: string };
+
+    /**
+     * The image used to start a container.
+     */
+    readonly image: ecs.EcrImage;
```

Comment:
I believe you want `ecs.ContainerImage` here (unless Batch only works with ECR images for some reason...?).

Reply:
Hm, I see now you are right. But I am a bit stumped here on how to bind the `ContainerImage` to this construct. Could you assist me here on this one (or point me to how...?)

Reply:
Hmm, I think you might be right here.

OK, let's leave it as `ecs.EcrImage` then.

Reply:
I noticed that there is a way to do this but requires us to extend the `ecs.ContainerDefinition` construct. But this would drastically misinterpret the usage of the Batch `JobDefinition` since it doesn't actually have all the same properties available.

However, I am curious. We might be fine using it anyways if it is possible to use the underlying `TaskDefinition` construct for a `JobDefinition` as well and not run into issues in Cloudformation.

What are your thoughts then?

Reply:
I think you lost me here... Can you elaborate what you mean by "We might be fine using it anyways if it is possible to use the underlying `TaskDefinition` construct for a `JobDefinition` as well and not run into issues in Cloudformation."? Maybe with some (pseudo-, if necessary) code?

Reply:
I am saying that in order to call `.bind(...)` on a `ContainerImage`, we need to have the `JobDefinition` extend or match the `ContainerDefinition` construct. However, that construct contains way more properties than we need for this.

So, my thought here is we pseudo fix this. We make a private class that extends the `ContainerDefinition` construct to allow transposing our job definition container. This way we can allow the use of `ContainerImage` when providing an image.